driver/src/contract.rs

```diff
@@ -227,13 +231,26 @@ impl SnappContract for SnappContractImpl {
             options.gas = Some(U256::from(5_000_000));
             self.contract.call(
                 "applyAuction",
-                (slot, prev_state, new_state, order_hash, prices_and_volumes),
+                (slot, prev_state, new_state, order_hash, standing_order_index, prices_and_volumes),
                 account,
                 options,
             ).wait()
             .map_err(DriverError::from)
             .map(|_|())
     }
+    
+    fn calculate_order_hash(
```

Comment:
Are you sure we want to use a smart contract function here? Shouldn't we compute this hash based on the information we got from the database ourselves? Otherwise we might pass invalid order data to the solver without noticing.

Reply:
Sure, this would be the second step. New Pr just arrived.

Reply:
I wanted to split the PRs for an easier review...

Reply:
Ok, I think you could have also put the other PR first (exposing the native order hash calculation) and then just use it here. I feel it's a bit of a waste to add a new function here, just to immediately remove it again.